controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -291,8 +286,8 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	}
 
 	readyMachines := int32(0)
-	for _, m := range ownedMachines {
-
+	for i := range ownedMachines.Items {
+		m := &ownedMachines.Items[i]
```

Comment:
Curious if this change is necessary or preference?

Reply:
Good question! I think it would be necessary if anything held the reference once the iteration completed, because of how [range works](https://github.com/golang/go/wiki/Range#gotchas). But I believe nothing holds the reference.

However, the change _does_ resolve this linter warning:
> Using a reference for the variable on range scope `m` (scopelint)